common/collection/interface.go

```diff
@@ -28,6 +28,8 @@ type (
 		// Add push an item to the queue
 		Add(item interface{})
 		// Remove pop an item from the queue
+		// TODO: consider return an error here to indicate no more item in the Queue
```

Comment:
If you add a nil check when adding new items, then remove can also return nil

Reply:
Makes sense. Added the check when adding new items. Removed the TODO here.